packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -93,6 +101,34 @@ export function unstable_httpSubscriptionLink<
           };
           // console.log('starting', new Date());
           eventSource.addEventListener('open', onStarted);
+
+          eventSource.addEventListener('open', () => {
+            observer.next({
+              result: {
+                type: 'state',
+                state: 'pending',
+              },
+            });
+          });
+
+          eventSource.addEventListener('error', () => {
+            if (eventSource?.readyState === EventSource.CLOSED) {
+              return;
+            }
+
+            const error = TRPCClientError.from(
+              new Error(`EventSource Network Error`),
+            );
```

Comment:
We can probably use some sort of value from the callback here

Reply:
Couldn't find anything in the typedef of the event, but I haven't looked that hard into it, console does say CONNECTION_DISCONNECTER or something similar, so there might be something to extract?

Reply:
Yeah I realised we don't serialize errors here & it'd be quite involved to do

Reply:
The Event that gets emitted can be of type ErrorEvent. Resolving this :)

Reply:
Nevermind, when using Chrome it just seems to be an Event and not an ErrorEvent, haven't tested on other browsers